### PR TITLE
added speed check to locationlistener

### PR DESCRIPTION
**Pull request by Samkeer1**
Relates to https://github.com/revelry-foundation/nightingale/issues/52
Added a speed check to the 'locationlistener', to simply not save locations where a User was moving 10 or more MPH.

I compared recorded locations with a list of all locations, and only speeds recorded at under 10mph were saved.
Screen Shot 2020-06-04 at 2 39 56 PM

Possible note for a future ticket:
For the sake of saving power, it looks like we have the tools to stop or start tracking location on 'activity change', with activities including 'on_foot', 'still', 'in_vehicle', and possibly more. 
However, the IOS xcode simulator seems to be producing 'unknown' activity results.

### CODE REVIEW

File: src/components/LocationListener.tsx [hunk 1]
```diff
@@ -72,13 +72,16 @@ class LocationListener extends Component<Props> {
     const {
       coords: {latitude, longitude},
       timestamp,
+      speed,
```
Comment (NOrzechowski) [inline src/components/LocationListener.tsx:75]: As far as I can tell there is no `speed` property coming from `Location`. Am I missing something?

Comment (NOrzechowski) [reply src/components/LocationListener.tsx:75]: I think it's an object within `coords` so if you move it there it should be fine

Comment (Samkeer1) [reply src/components/LocationListener.tsx]: Good catches. Thank you

File: src/components/LocationListener.tsx [hunk 2]
```diff
@@ -70,15 +70,16 @@ class LocationListener extends Component<Props> {
...
+      coords: {latitude, longitude, speed},
       timestamp,
     } = location
-
-    this.props.locationStorage.recordLocation({
-      latitude: latitude,
-      longitude: longitude,
-      when: timestamp,
-    })
+    if (speed && speed <= 10 * 2.23694 || !!speed == false) {
```
Comment (NOrzechowski) [inline src/components/LocationListener.tsx:76]: two things: can you extract the magic number into a constant and `|| !!speed == false` seems really weird to me. can you write out in a comment  exactly what this predicate is doing?

File: src/components/LocationListener.tsx [hunk 3]
```diff
@@ -70,15 +70,20 @@ class LocationListener extends Component<Props> {
...
+    const MPS_TO_MPH_RATE = 2.23694
+    const speedInMph = speed && speed * MPS_TO_MPH_RATE // meters/s -> mph
 
-    this.props.locationStorage.recordLocation({
-      latitude: latitude,
-      longitude: longitude,
-      when: timestamp,
-    })
+    // only recording locations user was moving under 10 mph at
+    if (speedInMph && speedInMph <= 10 || !!speed == false) { // !!speed coerces value of speed to a boolean, to account for null, undefined, 0, all which we want to record by default
```
Comment (NOrzechowski) [inline src/components/LocationListener.tsx:80]: I guess i'm still hung up on `!!speed == false`. Aren't you alreading checking if speed exists earlier in the predicate where you say `speedInMph && ...`? Also, if it's definitely a boolean I think you'd want ` === false` or just `!speed`? I think we should pair on this to help alleviate my confusion.

**Pull request merged**